core/src/main/java/com/linecorp/armeria/client/HttpChannelPool.java

```diff
@@ -118,6 +124,14 @@ protected void initChannel(Channel ch) throws Exception {
                                                       .get(ChannelOption.CONNECT_TIMEOUT_MILLIS);
     }
 
+    private static void addSocksProxyHandlerIfPossible(
+            ChannelPipeline pipeline, Optional<? extends ProxyHandler> proxyHandler) {
+        if (proxyHandler.isPresent() && (proxyHandler.get() instanceof Socks4ProxyHandler ||
```

Comment:
Maybe 1 more HttpProxyHandler for CONNECT?

Reply:
I think `HttpProxyHandler` should be added after ssl is applied (at least that's what the netty examples were doing 🤔 )

Reply:
You mean between client and proxy using https? If we use HttpProxyHandler directly, we may support nomral HTTP CONNECT one I think.

--
Also can you tell me the link, just curious about which example.

Reply:
https://github.com/netty/netty/blob/5c458c9a98d4d3d0345e58495e017175156d624f/handler-proxy/src/test/java/io/netty/handler/proxy/ProxyHandlerTest.java#L360

Didn't really take a thorough look though..

Reply:
So it looks like 2 cases.
If we want to provide https proxy, I think as you said we need to put before HttpProxyHandler. 
But there is another https between client and the real server I think.

Reply:
Ah,   I see.  You're right in that there are 2 cases.
This may complicate the api a bit (we may need to add another ssl handler if we want to support `Http[s]ProxyHandler` 🤔 )